lsp-semantic-tokens.el

```diff
@@ -670,22 +670,24 @@ IS-RANGE-PROVIDER is non-nil when server supports range requests."
                        (lsp-warn "No face has been associated to the %s '%s': consider adding a corresponding definition to %s"
                                  category id varname)) maybe-face)) identifiers)))
 
-(defun lsp-semantic-tokens--replace-alist-values (a b)
-  "Replace alist A values with B ones where available."
-  (-map
-   (-lambda ((ak . av))
-     (cons ak (alist-get ak b av nil #'string=)))
-   a))
+(defun lsp-semantic-tokens--apply-alist-overrides (base overrides)
+  "Merge alist BASE with OVERRIDES.
+For keys present in both alists, the assignments made by
+OVERRIDES will take precedence."
+  (let ((all-keys (-union (mapcar #'car base) (mapcar #'car overrides))))
```

Comment:
Something like that seems to be more natural:

```
(let* ((copy-base (copy-alist base)))
  (mapc (-lambda ((key . value)) (setf (alist-get key copy-base) value)) overrides)
  copy-base) 
```

Reply:
How I tested it:

```
(let* ((base '((key1 . value1) (key2 . value2)))
       (overrides '((key1 . override1)
                    (new-key . override2)))
       (copy-base (copy-alist base)))
  (mapc (-lambda ((key . value)) (setf (alist-get key copy-base) value)) overrides)
  copy-base)
```

Reply:
thanks!